x/mongo/driver/topology/connection_test.go

```diff
@@ -485,6 +485,23 @@ func TestConnection(t *testing.T) {
 					}
 					listener.assertMethodsCalled(t, 1, 1)
 				})
+				t.Run("message too large errors", func(t *testing.T) {
```

Comment:
Great test! There should also be a test to assert that the 48MB default is used. It should be pretty similar to this one, but you'll have to change the `buffer` to contain a length greater than 48MB and change the connection's description to be empty.

For this kind of thing, I'd recommend using table-driven tests: https://github.com/golang/go/wiki/TableDrivenTests.

Reply:
Sure! Moved this test, and the 48MB default one you described to a new table-driven test `TestReadWireMessageTooLarge`.